### ISSUE TITLE
calcite-slider > decouple ticks, labels and stops

### ISSUE THREAD

**Issue by richiecarmichael** [Esri/calcite-components#5827]
### Description

At present `calcite-slider` assumes that ticks, tick labels and stops coincide _and_ that all three are at regular intervals. This is rarely the case. The solution is to have independent properties for all three. For examples:

```js
{
  min: 0,
  max: 100,
  // Place small ticks at 10 unit spacing
  ticks: [0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100],
  // Label each quarter.
  labels: [0, 25, 50, 75, 100], 
  // Slider thumbs will snap at 10 unit intervals.
  steps: [0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100]
}
```

###  What are Labels?
Label will consist of text and longer tick mark, See #2584 for proposal on custom label formatter.

### Design
```ts
class slider {
  ticks: number[];
  labels: number[];
  steps: number[];
}
```

### Breaking changes:
1. `slider.labelTicks` is obsolete
2. `slider.ticks` changes from `number` to `number[]`

// @driskull @jcfranco

### Acceptance Criteria

See above.

### Related issues:
#5522 - **calcite-slider > layout vertical and horizontal**
#2584 - **Enhancement: calcite-slider - custom tick label formatters**
#1631 - **Enhancement: Configurable slider highlighting**

### Which Component

calcite-slider

### Example Use Case

![image](https://user-images.githubusercontent.com/2531121/204422295-41e35fd0-d278-49b1-906d-a4debf8ccc65.png)

### Esri team

ArcGIS API for JavaScript

**Comment by ubatsukh** [comment]
@driskull can you please change the priority of this issue to high as MV is asking for dimension/range slider. @jcfranco, we hope you can work the slider issues when you get back from your paternity leave. :) 

The other slider issues that need to be addressed before the Ranger slider widget can be implemented in the API: 
https://github.com/Esri/calcite-components/issues/5522
https://github.com/Esri/calcite-components/issues/2584
https://github.com/Esri/calcite-components/issues/1631

Here are the API requirements for the Range Slider: https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/46032#issuecomment-3805946

**Comment by driskull** [comment]
@ubatsukh marked them as higher priority. We will figure out which milestone these can be taken care of after the 1.0 release.

**Comment by brittneytewks** [comment]
Better understand user priority. Start in April for future milestone implementation

**Comment by geospatialem** [comment]
Priority has shifted to a medium status, with Maps SDK shifting priority to #6590 and #6591.

**Comment by driskull** [comment]
Since this is a feat/refactor with possible breaking change should it be pushed to July or Oct?

**Comment by geospatialem** [comment]
Per [Matt's comment above](https://github.com/Esri/calcite-components/issues/5827#issuecomment-1601778666) removing from the June milestone, and will assign a milestone once the breaking change milestone is determined. cc @brittneytewks

**Comment by jcfranco** [comment]
Apologies for the belated update, but I think I have an idea on how to implement this without introducing a breaking change. 

This is what it could look like:

```ts
class Slider {
  ticks: number | number[]; 
  labelTicks: number | number[];
  step: number | number[];
}
```

Worth noting that we cannot rely entirely on arrays for these properties because they would prevent users from setting these in markup (we don't support passing rich data as attributes, see https://web.dev/articles/custom-elements-best-practices#aim-to-only-accept-rich-data-objects,-arrays-as-properties for more info).

I'm planning to sync up with @richiecarmichael and also have some additional time for testing. With that said, this would have to be pushed to December.

**Comment by jcfranco** [comment]
Bumped the priority on this one. cc @brittneytewks @geospatialem

**Comment by geospatialem** [comment]
Additional efforts are needed to address prior to landing the fix, which are now targeted for the January 2024 release.

**Comment by jcfranco** [comment]
Blocked by https://github.com/Esri/calcite-design-system/issues/5522.

**Comment by richiecarmichael** [comment]
Good progress has been made to decouple steps, ticks and labels. Steps and ticks are now independent and while labels are coincident with ticks can be customized with a `labelFormatter`.

However steps and ticks are still regularly spaces from the start of the slider. When dealing with temporal data, irregular spaced steps, ticks and labels are the norm. Below are a few scenarios that have irregular steps/ticks/labels:
1. A slider with ticks representing imagery captured during the day.
2. A slider representing a time range with steps/ticks/labels representing:
  a) months (the number of days per month varies)
  b) years (leap years, leap seconds)

In [this](https://codepen.io/richiecarmichael/pen/xxvbGZG?editors=1000) sample, the months of 2024 are shown as steps and ticks. Note that because these are evenly spaced that the start of July is assumed to be July 2.
![image](https://github.com/user-attachments/assets/2a34e43c-ae2c-4045-b6a9-4329de7cdb5c)

**Comment by DintaMel** [comment]
Per my conversation with @richiecarmichael and @ubatsukh, this is lower priority given the introduction of the [arcgis-slider](https://developers.arcgis.com/javascript/latest/references/common-components/components/arcgis-slider/). 

They do still want these changes eventually to support the transition from the [arcgis-time-slider](https://developers.arcgis.com/javascript/latest/references/map-components/components/arcgis-time-slider/) to Calcite's Slider. 

Additionally, since designs must often be mobile friendly, the need for tick marks has decreased.
